### PR TITLE
Add tests for getColumns

### PR DESCRIPTION
**Pull request by ConorNeo**

### CODE REVIEW

File: neo4j-jdbc/src/main/java/org/neo4j/driver/jdbc/DatabaseMetadataImpl.java [hunk 1]
```diff
@@ -941,39 +941,41 @@ public ResultSet getColumns(String catalog, String schemaPattern, String tableNa
...
-				values[18] = Values.NULL; // SCOPE_CATALOG
-				values[19] = Values.NULL; // SCOPE_SCHEMA
-				values[20] = Values.NULL; // SCOPE_TABLE
-				values[21] = Values.NULL; // SOURCE_DATA_TYPE
-
-				rows.add(values);
+			if (!propertyName.isNull() && !propertyTypes.isNull()) {
+				var propertyTypeList = propertyTypes.asList(propertyType -> propertyType);
+				if (propertyTypeList.size() > 1) {
+					// some kind of logging?-
```
Comment (michael-simons) [inline neo4j-jdbc/src/main/java/org/neo4j/driver/jdbc/DatabaseMetadataImpl.java:947]: I added a bunch of those, why not do something similar?

**Review by michael-simons — APPROVED** [review]

File: neo4j-jdbc/src/main/java/org/neo4j/driver/jdbc/DatabaseMetadataImpl.java (hunk 1 above)
Comment (ConorNeo) [reply neo4j-jdbc/src/main/java/org/neo4j/driver/jdbc/DatabaseMetadataImpl.java:947]: oh nice i was meant to ask about it, ill add them quickly before I merge

**Comment by ConorNeo** [comment]
I found some bigger issues. Currently, the types returned by the procedure used (db.schema.nodeTypeProperties()) are Java types... This is a bug and will be fixed by 6; what that means now is that the mapping needs to be changed so that we are expecting the "java-y" types (String, not STRING) and convert them to their cypher type counterpart for the TYPE_NAME column, because this expects driver specific types, so cypher types. Furthermore will need map this java-y type to the correct SQL type for the DATA_TYPE column. The latest commit is WIP and is the start, and the tests will need updating. Should get to it Monday before meetings.

**Comment by michael-simons** [comment]
> This should do for now, but I think arrays and structures might make this a little more complex as, whilst it won't break, how valid is a STRING with an array in it?

Maybe in some theoretical setup, but not in the JDBC world…

**Pull request merged**